packages/design-tokens/scripts/style-dictionary.js

```diff
@@ -0,0 +1,495 @@
+/**
+ * Ported from primer/primitives, as it was removed in v8 release and we still need it
+ */
+
+const glob = require('fast-glob')
+const StyleDictionary = require('style-dictionary')
+
+const {fileHeader, formattedVariables} = StyleDictionary.formatHelpers
+
+//-----
+// functions to be extracted
+// TODO: extract to a separate files
+
+const pathToKebabCase = token => token.path.join('-')
+
+const pathToDotNotation = token => token.path.join('.')
+
+const capitalize = string => string[0].toUpperCase() + string.slice(1)
+
+const pathToPascalCase = token => token.path.map(tokenPathItems => capitalize(tokenPathItems)).join('')
+
+// REGISTER THE CUSTOM TRANFORMS
+
+/**
+ * transform: scss variable names
+ * example: `$namespace-item-variant-property-modifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/scss',
+  type: 'name',
+  transformer: pathToKebabCase,
+})
+
+/**
+ * transform: css variable names
+ * example: `--namespace-item-variant-property-modifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/css',
+  type: 'name',
+  transformer: pathToKebabCase,
+})
+
+/**
+ * transform: js variable names
+ * example: `namespace.item.variant.property.modifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/js',
+  type: 'name',
+  transformer: pathToDotNotation,
+})
+
+/**
+ * transform: js es6 variable names
+ * example: `NamespaceItemVariantPropertyModifier`
+ */
+StyleDictionary.registerTransform({
+  name: 'name/js/es6',
+  type: 'name',
+  transformer: pathToPascalCase,
+})
+
+// find values with px unit
+function isPx(value) {
+  return /[\d.]+px$/.test(value)
+}
+
+// transform: px to rem
+StyleDictionary.registerTransform({
+  name: 'pxToRem',
+  type: 'value',
+  transformer: token => {
+    if (isPx(token.value)) {
+      const baseFontSize = 16
+      const floatValue = parseFloat(token.value.replace('px', ''))
+
+      if (isNaN(floatValue)) {
+        return token.value
+      }
+
+      if (floatValue === 0) {
+        return '0'
+      }
+
+      if (token.name.includes('lineHeight')) {
+        return `${floatValue / baseFontSize}`
+      }
+
+      return `${floatValue / baseFontSize}rem`
+    }
+    return token.value
+  },
+})
+
+//-----
+
+// ts output
+StyleDictionary.registerTransform({
+  name: 'attribute/typescript',
```

Comment:
How does this work? Interesting. Can you explain it?

Reply:
![image](https://github.com/user-attachments/assets/20aeaad8-c845-4b41-a96b-a095c898f46f)

Reply:
I'll get back to you on that... 😅 

...added as a follow up PR item.